program/src/instructions/collect.rs

```diff
@@ -22,10 +22,13 @@ pub struct Collect<'info> {
     // n fee accounts in remaining accounts
     // where the accounts are derived from the passed in
     // indices and bumps, in order
+
+    // TODO: I guess we're okay with having no signers
 }
 
 pub fn process_collect<'info>(
     ctx: Context<'_, '_, '_, 'info, Collect<'info>>,
+    // TODO: curious, why do we pass this in as an argument rather than having an internal state? Just because we dont know how many shards there will be?
```

Comment:
Goal was to simplify by having no state. No need to initialize, transfers to/from show up as CPIs into the system program so easier to index/log, etc.